builtin/logical/pki/path_acme_new_account.go

```diff
@@ -256,20 +274,22 @@ func (b *backend) acmeNewAccountSearchHandler(acmeCtx *acmeContext, r *logical.R
 }
 
 func (b *backend) acmeNewAccountCreateHandler(acmeCtx *acmeContext, r *logical.Request, fields *framework.FieldData, userCtx *jwsCtx, data map[string]interface{}, contact []string, termsOfServiceAgreed bool) (*logical.Response, error) {
-	if userCtx.Existing {
-		return nil, fmt.Errorf("cannot submit to newAccount with 'kid': %w", ErrMalformed)
-	}
-
 	// If the account already exists, return the existing one.
 	if b.acmeState.DoesAccountExist(acmeCtx, userCtx.Kid) {
+		// TODO: We should possibly update account values here such as contact.
 		return b.acmeNewAccountSearchHandler(acmeCtx, r, fields, userCtx, data)
 	}
 
-	// TODO: Limit this only when ToS are required by the operator.
-	if !termsOfServiceAgreed {
-		return nil, fmt.Errorf("terms of service not agreed to: %w", ErrUserActionRequired)
+	if userCtx.Existing {
+		return nil, fmt.Errorf("cannot submit to newAccount with 'kid': %w", ErrMalformed)
```

Comment:
This is the case for the TODO above, IMO. It'll be a little weird, because we'll wanna give this error on `newAccount` paths, but not on `account-update` paths.

Reply:
I think we should also handle deactivation here: https://datatracker.ietf.org/doc/html/rfc8555#section-7.3.6

Reply:
Added support for deactivation within the new account update handler.